crates/sui-indexer/src/store/indexer_analytical_store.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use async_trait::async_trait;
+
+use crate::models_v2::address_metrics::{StoredActiveAddress, StoredAddress, StoredAddressMetrics};
+use crate::models_v2::checkpoints::StoredCheckpoint;
+use crate::models_v2::move_call_metrics::{StoredMoveCall, StoredMoveCallMetrics};
+use crate::models_v2::network_metrics::StoredNetworkMetrics;
+use crate::models_v2::transactions::StoredTransaction;
+use crate::models_v2::tx_count_metrics::StoredTxCountMetrics;
+use crate::models_v2::tx_indices::StoredTxIndex;
+use crate::types_v2::IndexerResult;
+
+#[async_trait]
+pub trait IndexerAnalyticalStore {
```

Comment:
I'm not sure i see a whole lot of value in this trait vs just passing the concrete type through to the handlers at this point in time.

Reply:
the idea is that, as it's tbd where the analytical codes will live in the long run, so I wanted to keep it separate from the existing indexer v2 implementation such that
- it's easier to move them around or just delete it if no longer needed
- it allows more flexibilities to run analytical things in a separate binary in case blocking threads compete